### ISSUE TITLE
Add matplotlib example to the user guide?

### ISSUE THREAD

**Issue by stefanv** [Zulko/moviepy#421]
Thanks for MoviePy!

I was just showing it to a colleague, and realised that the matplotlib example is hidden in a blog post. The common use cases shown here:

http://zulko.github.io/blog/2014/11/29/data-animations-with-python-and-moviepy/

would make for a great addition to the user guide.

**Comment by kerstin** [comment]
Thanks for the suggestion, @stefanv!

@Zulko is pretty busy, so not sure when (if) he'll find time to add his own examples. The project accepts pull requests though, so if you wanted to add examples to the docs, that'd be great!

**Comment by stefanv** [comment]
I'll wait for #368 to be resolved before adding the examples.

**Comment by flothesof** [comment]
Hi @kerstin and @stefanv 

Since I often use moviepy to do matplotlib animations, I'd like to look into adding matplotlib to documentation.
This is where I need some input. The documentation is, so far, organized like this:

- getting started
- gallery 
- example scripts

It turns out that the "gallery" section links to the matplotlib examples from Zulko's blog post. 
So I could expand the section in the gallery part and add matplotlib examples to it. 
However, we could also add a special section since matplotlib has a large user base.
To not disrupt things, I'll go with the first version. Let me know if you think it's better to make a separate page.

On another note, when I open [http://zulko.github.io/moviepy/gallery.html](http://zulko.github.io/moviepy/gallery.html) in my browser, the page downloads lot of small files that look like YouTube links. Do you observe the same thing? Does anyone know how to fix this?

Best regards,
Florian

(pull request to be followed later, hopefully)

**Comment by kerstin** [comment]
@flothesof Just a quick note on your browser prompting you to download stuff: looks like I only fixed video links with #446 but there's still mixed (secure + insecure) content because of images prefixed with `http` instead of `https`. Will fix image links next.

Regarding adding examples: I think I'd also just extend the existing sections for now. If one particular kind becomes so big it could get its own separate section, I guess we could still split it off.

**Comment by kerstin** [comment]
@flothesof The image embeds should be fixed now.

**Comment by kerstin** [comment]
Well, this is weird. It looks like none of my changes are reflected in the docs... hmm.

**Comment by kerstin** [comment]
My bad, I thought GitHub autogenerates documentation like Gitlab does. I'm getting lots of errors and warnings when trying to generate the docs though, and end up with pages missing content or not being there at all.

I tried to figure out what the problem could be with help of the [Sphinx docs](http://www.sphinx-doc.org/en/stable/index.html) but haven't made any progress so far. (Also, I guess this should get moved to a separate issue.)

**Comment by stefanv** [comment]
@flothesof I agree that expanding the existing sections is the best way forward for now.  I'm waiting on @kerstin's changes to reflect to test whether the file downloads are now gone.

**Comment by flothesof** [comment]
Hi all,

so here's a pull request that integrates the basic example from Zulko's blog post and also adds a note about Jupyter Notebook usage #465.
Let me know what you think.
Best,
Florian

**Comment by flothesof** [comment]
@kerstin, regarding how to update the documentation, I think you should build the docs locally and then push them to the gh-pages branch of the repo. This should then be displayed when one opens the link to the docs [http://zulko.github.io/moviepy/](http://zulko.github.io/moviepy/).

**Comment by Zulko** [comment]
@flothesof I think she got that, but apparently the docs need some cleaning (lots of errors and warning when you build). Have you tried compiling and if yes, have you noticed the same ? Thanks for the PR btw, it looks good.

**Comment by flothesof** [comment]
Hi @Zulko,
thanks for your feedback. Yes, I've built the docs locally to see what they generate. I also have lots of errors. Some samples: 
```
Failed to import 'moviepy.video.tools.tracking': no module named moviepy.video.tools.tracking
[...]
WARNING: [autosummary] failed to import 'moviepy.video.fx.all.rotation': no module named moviepy.video.fx.all.rotation
[...]
/Users/kappamaki/workspace/moviepy/docs/ref/videofx/moviepy.video.fx.all.freeze_at_end.rst:6: WARNING: autodoc: failed to import function 'freeze_at_end' from module 'moviepy.video.fx.all'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/kappamaki/anaconda/lib/python3.5/site-packages/sphinx/util/inspect.py", line 109, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: module 'moviepy.video.fx.all' has no attribute 'freeze_at_end'
```
Best,
Florian

**Comment by kerstin** [comment]
I created an issue regarding the documentation needing fixing: #482

**Comment by mbeacom** [comment]
@stefanv #465 has been merged and we're preparing to push the updated documentation. Can this be closed?

**Comment by stefanv** [comment]
Thanks!
